src/deduper.rs

```diff
@@ -183,20 +189,35 @@ fn write_attributes(
                     attributes[&cfg.attribute_name] = Value::Array(Vec::new());
                 } else {
                     let dedupe_key = VecDeque::from([document_key.as_str()]);
-                    let hashes = bloom_filter.hashes(&dedupe_key);
-                    if bloom_filter.contains(&hashes) {
-                        // attributes[&cfg.attribute_name] = Value::Bool(true);
-
-                        let mut duplicate_docs_array = Vec::new();
-                        let attr = vec![
-                            Value::from(0),
-                            Value::Number(document_key.len().into()),
-                            Value::from(1),
-                        ];
-                        duplicate_docs_array.push(Value::Array(attr));
-                        attributes[&cfg.attribute_name] = Value::Array(duplicate_docs_array);
-                    } else if !bloom_filter.read_only {
-                        bloom_filter.insert(&hashes);
+
+                    //Just compute the first hash to see if it matches the partition
+                    let mut hashes = vec![bloom_filter.first_hash(&dedupe_key)];
+                    num_observed += 1;
+                    if in_partition(
+                        hashes[0],
+                        dedupe_config.num_partitions.unwrap_or(1),
+                        dedupe_config.partition_index.unwrap_or(0),
+                    ) {
+                        num_processed += 1;
+                        //Compute the remaining hashes
+                        hashes.extend(bloom_filter.remaining_hashes(&dedupe_key));
+                        if bloom_filter.contains(&hashes) {
```

Comment:
uhm I don't fully follow the logic here. What I would do is have this `contains` function check if whatever you are putting into the bloom filter matches partition, and if so, put that into the filter.